### PR TITLE
Fix/68 transaction processor

### PR DESCRIPTION
**Pull request by florianlenz**
#### Changes
- changed transaction processor to return promise

#### Verification
- ci should pass

closes: #68

### CODE REVIEW

File: src/queues/transaction.js [hunk 1]
```diff
@@ -271,19 +269,13 @@ export default class TransactionQueue implements TransactionQueueInterface {
                     }, (err) => {
                         // @todo handle error (maybe log?)
                         if (err) {
-                            console.log(err);
+                            return rej(err);
                         }
-                        this._ee.emit(TRANSACTION_QUEUE_FINISHED_CYCLE);
-                        setTimeout(processAction, this._processingTimeout);
+                        setTimeout(res, this._processingTimeout);
```
Comment (seland) [inline src/queues/transaction.js]: Why should we call success callback after timeout if there is an error?

Comment (florianlenz) [reply src/queues/transaction.js]: `res` will resolve the promise | `rej` will resolve it. It look's good for me ...

Comment (seland) [reply src/queues/transaction.js]: Ah, sorry, it’s fine. Do you use timeout here intentionally, so one won’t call it too frequently?

Comment (florianlenz) [reply src/queues/transaction.js]: yes

Comment (seland) [reply src/queues/transaction.js]: Okay then

**Pull request merged**